### ISSUE TITLE
Handle someone choosing to create snapcraft.yaml before name is registered

### ISSUE THREAD

**Issue by matthewpaulthomas** [canonical-ols/build.snapcraft.io#418]
We recommend that someone register the snap name before adding a snapcraft.yaml — because for publishing, snapcraft.yaml needs to include the registered name.

But you might want to add a snapcraft.yaml before registering the snap name, for example if you want to see whether your software works as a snap at all.

Therefore, we should customize the contents of the template snapcraft.yaml in this case, to tell you that it won’t publish until you register the name, and tell you how to proceed.

**Comment by cjwatson** [comment]
Is that really an appropriate thing to put in the template `snapcraft.yaml`?  I predict that, a year down the line, this will result in many `snapcraft.yaml` files lying around people's repositories that document a required action that they took eleven months ago.

**Comment by matthewpaulthomas** [comment]
This was a request from @evandandrea, so I may not have the details exact.

One approach might be to make it a one-line comment-within-a-comment:

`# name: # Uncomment once you've registered a name at build.snapcraft.io (or using 'snapcraft register')`

Another approach might be just to not worry about the clutter. I haven’t coded in enough languages/frameworks to make analogies here, but I wouldn’t be surprised if there are many examples of repos with boilerplate text in their config files. One example I find with quick Googling is thousands of repos that claim to contain “[a sample Guardfile](https://www.google.co.uk/search?q=%22a+sample+guardfile%22+site:github.com+-site:gist.github.com)” when the Guardfile has actually been customized for the project.

**Comment by cjwatson** [comment]
Right, I appreciate that there are lots of cases where other people have ended up with clutter; I'm just trying to encourage avoiding it here if possible.  Another approach might be to see if we can present this advice out-of-band instead when we're directing the user to write `snapcraft.yaml`?

**Comment by evandandrea** [comment]
Today on build.snapcraft.io you can add a snapcraft.yaml from a template
before you register the name. We even kick off builds that continue to
publishing when you later register the name.

That template currently makes the name "my-snap-name" with a comment of
"you probably want to 'snapcraft register <name>".

No one is going to be able to own "my-snap-name".

**Comment by matthewpaulthomas** [comment]
If we were providing our own snapcraft.yaml editor (#390), we could give the advice outside the template. But while the template is in the github.com editor, in a new window/tab, I don’t see how we can.

[Spec updated](https://docs.google.com/document/d/1koTwI3cRr2S2vj5zHKKbR7aWxLRfShDVq2gHQ29cDyU/edit#heading=h.30j8f330tbd6): “If you have already registered a name for this snap, the template should use that name; otherwise, it should contain ‘`# name: Uncomment this after registering a name on build.snapcraft.io.`’.” I’m open to suggestions for better text.

**Comment by matthewpaulthomas** [comment]
[Revised text](https://docs.google.com/document/d/1koTwI3cRr2S2vj5zHKKbR7aWxLRfShDVq2gHQ29cDyU/edit#heading=h.30j8f330tbd6):
> # After registering a name on build.snapcraft.io, commit an uncommented line:
> # name: your-registered-name

**Comment by nottrobin** [comment]
I don't know if this is still a problem in the new snapcraft.io/build experience. I'm moving it to snapcraft.io repo, it can be decided there.
